benchmark/main.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+$:.push File.expand_path("lib", __dir__)
+
+require "benchmark"
+
+ENV["RAILS_ENV"] = "test"
+
+require "active_record"
+require "goldiloader"
+
+db_adapter = ENV.fetch('ADAPTER', 'sqlite3')
+db_config = YAML.safe_load(File.read('spec/db/database.yml'))
+ActiveRecord::Base.establish_connection(db_config[db_adapter])
+
+require_relative "./../spec/db/schema.rb"
+
+# Setup data
+blog_1 = Blog.create!
+blog_2 = Blog.create!
+100.times do
+  user_1 = User.create!
+  user_2 = User.create!
+  Post.create!(author: user_1, blog: blog_1)
+  Post.create!(author: user_2, blog: blog_2)
+end
+
+Benchmark.bm(20) do |x|
+  # Use AR's eager loading
+  x.report("AR eager loading: ") do
+    ::Blog.all.includes(posts: :author).map do |blog|
+      blog.posts.to_a.each do |post|
```

Comment:
I'm not sure it matters but this benchmark will generate less garbage if we drop the `to_a` and call `blog.posts.each`.